### ISSUE TITLE
🐛 Biome encountered an unexpected error

### ISSUE THREAD

**Issue by oaleynik** [biomejs/biome#5400]
### Environment information

```block
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-ghostty"
  JS_RUNTIME_VERSION:           "v23.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/10.6.4"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

Just ran `pnpx @biomejs/biome check src/`. After a couple of reported issues, the message has been displayed that Biome encountered an unexpected error.

```
Source Location: /Users/runner/work/biome/biome/crates/biome_text_size/src/range.rs:66:9
Thread Name: main
Message: assertion failed: start.raw <= end.raw
```

biome.json:
```json
{
  "$schema": "https://biomejs.dev/schemas/1.9.4/schema.json",
  "vcs": {
    "enabled": true,
    "clientKind": "git",
    "useIgnoreFile": true
  },
  "files": {
    "ignoreUnknown": true,
    "ignore": [
      "app/**/*.css",
      "app/third-party",
      "docker",
      "package-lock.json",
      "pnpm-lock.yaml",
      "public"
    ]
  },
  "formatter": {
    "enabled": true,
    "indentStyle": "space",
    "indentWidth": 2
  },
  "organizeImports": {
    "enabled": true
  },
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": true,
      "complexity": {
        "noForEach": "off"
      },
      "style": {
        "recommended": false
      },
      "suspicious": {
        "noExplicitAny": "off",
        "noAssignInExpressions": "off"
      }
    }
  },
  "javascript": {
    "formatter": {
      "quoteStyle": "single"
    }
  }
}
```

### Expected result

The linter check finishes without crashing.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by oaleynik** [comment]
Looks like the issue is with useDefaultSwitchClauseLast. When I updated the switch, Biome's lint stopped crashing.

```
src/store/components/store.vue:65:5 lint/suspicious/useDefaultSwitchClauseLast ━━━━━━━━━━━━━━━━━━━━━

  ✖ The default clause should be the last switch clause.

    63 │ const paymentProviderURL = computed(() => {
    64 │   switch (userMerchantType.value) {
  > 65 │     default:
       │     ^^^^^^^^
  > 66 │       return 'https://dashboard.stripe.com/';
       │       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    67 │     case 'paypal':
    68 │       return 'https://paypal.com/';

  ℹ The following case clause is here:

    17 │         v-if="merchantDisabled && paymentProvider !== 'None' && !merchantKyc"
    18 │         :message="`Your store is deactivated until you contact ${paymentProvider}`"
  > 19 │         :button-text="paymentProvider + ' dashboard'"
       │                                       ^^^^^^^^^^^^^^^
  > 20 │         :target="'_blank'"
  > 21 │         :href="paymentProviderURL"
```

**Comment by arendjr** [comment]
I'm afraid this is another casualty of our partial Vue support :( See: https://github.com/biomejs/biome/issues/4593#issuecomment-2583145999
